### ISSUE TITLE
Consider whether Citus daemons run on Follower Coordinators?

### ISSUE THREAD

**Issue by aamederen** [citusdata/citus#1575]
When implementing background workers, we should consider the existence of secondary coordinators and secondary workers.

For example, in a high-availability setup of a citus cluster, the secondary coordinator, as well as the primary coordinator, runs the deadlock detection logic periodically, i.e. both coordinators run `dump_local_wait_edges` on workers.

**Comment by onderkalaci** [comment]
> I think we should fix it by not running deadlock detection if the coordinator is in recovery mode.

@marcocitus does that make sense to you as well?

**Comment by marcocitus** [comment]
Are there cases in which we could have a distributed deadlock on a follower cluster? In that case we 
would want to run deadlock detection on a secondary. (probably not)

**Comment by marcocitus** [comment]
Going through some of the relevant [docs](https://www.postgresql.org/docs/current/static/hot-standby.html) and code, it seems distributed deadlocks could theoretically exist in follower clusters, but are probably edge cases that we can ignore (e.g. advisory locks). I think disabling deadlock detection in recovery mode makes sense.

**Comment by marcocitus** [comment]
We might want to run deadlock detection on a secondary coordinator in the default cluster.

**Comment by aamederen** [comment]
> We might want to run deadlock detection on a secondary coordinator in the default cluster, since it can perform certain distributed write transactions

Makes sense, but until we allow use of secondary coordinator on cloud, this will cause unnecessary traffic.
